File: examples/node/quickstart.mjs
// Quickstart: submit an event to the ingestion service via the JS SDK.
//
// Prereq: ingestion service running locally (cargo run -p heeczer-ingest).
//
// Run:
//   pnpm --dir bindings/heeczer-js install   # one-time
//   pnpm --dir bindings/heeczer-js build
//   node examples/node/quickstart.mjs
//
// Or import the published package once it's released:
//   import { HeeczerClient } from "@cognizhi/heeczer-sdk";

import { readFile } from "node:fs/promises";
import { fileURLToPath } from "node:url";
import { dirname, join } from "node:path";
import { HeeczerClient, HeeczerApiError } from "../../bindings/heeczer-js/dist/index.js";

const here = dirname(fileURLToPath(import.meta.url));
const eventPath = join(here, "..", "event.json");
const event = JSON.parse(await readFile(eventPath, "utf8"));

const baseUrl = process.env.HEECZER_BASE_URL ?? "http://127.0.0.1:8080";
const client = new HeeczerClient({
  baseUrl,
  apiKey: process.env.HEECZER_API_KEY,
});

console.log("» service version:", await client.version());

try {
  const { score, event_id } = await client.ingestEvent({
    workspaceId: "ws_default",
    event,
  });
  console.log(`» event ${event_id} ingested`);
  console.log(`» summary: ${score.human_summary}`);
  console.log(`» minutes=${score.final_estimated_minutes} band=${score.confidence_band}`);
} catch (err) {
  if (err instanceof HeeczerApiError) {
    console.error(`SDK error: kind=${err.kind} status=${err.status} message=${err.message}`);
    process.exit(1);
  }
  throw err;
}

File: examples/java/Quickstart.java
// Quickstart: submit an event to the ingestion service via the Java SDK.
//
// Prereq: ingestion service running locally in unauthenticated smoke-test mode
// (HEECZER_AUTH__ENABLED=false cargo run -p heeczer-ingest).
//
// Easiest path (JDK 17+ single-file source mode, JEP 458):
//   1. mvn -f bindings/heeczer-java install -DskipTests -Dgpg.skip=true
//   2. JAR=~/.m2/repository/com/cognizhi/heeczer-sdk/0.5.1-SNAPSHOT/heeczer-sdk-0.5.1-SNAPSHOT.jar
//      JACKSON=$(mvn -f bindings/heeczer-java -q exec:exec \
//        -Dexec.executable=echo -Dexec.args='%classpath')
//      java --class-path "$JAR:$JACKSON" examples/java/Quickstart.java
//
// Once the SDK is on Maven Central (plan 0012), Maven/Gradle coordinates will
// resolve Jackson transitively. The raw java --class-path invocation shown here
// still needs Jackson on the class-path explicitly.
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import io.github.cognizhi.heeczer.HeeczerApiException;
import io.github.cognizhi.heeczer.HeeczerClient;
import io.github.cognizhi.heeczer.IngestEventResponse;
import io.github.cognizhi.heeczer.VersionResponse;
import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;
import java.nio.file.Paths;

public final class Quickstart {

    public static void main(String[] args) throws IOException, InterruptedException {
        Path eventPath = Paths.get(System.getProperty("user.dir"), "examples", "event.json");
        ObjectMapper mapper = new ObjectMapper();
        JsonNode event = mapper.readTree(Files.readAllBytes(eventPath));

        String baseUrl = System.getenv().getOrDefault("HEECZER_BASE_URL", "http://127.0.0.1:8080");
        try (HeeczerClient client = new HeeczerClient.Builder(baseUrl)
                .apiKey(System.getenv("HEECZER_API_KEY"))
                .build()) {

            VersionResponse v = client.version();
            System.out.println("» service version: " + v.scoringVersion + "/" + v.specVersion);

            try {
                IngestEventResponse resp = client.ingestEvent("ws_default", event);
                System.out.println("» event " + resp.eventId + " ingested");
                System.out.println("» summary: " + resp.score.humanSummary);
                System.out.println("» minutes=" + resp.score.finalEstimatedMinutes
                        + " band=" + resp.score.confidenceBand);
            } catch (HeeczerApiException e) {
                System.err.println("SDK error: kind=" + e.getKind()
                        + " status=" + e.getStatus()
                        + " message=" + e.getMessage());
                System.exit(1);
            }
        }
    }
}
